### PR TITLE
Fix conflicts in `typing/printtyp.ml`

### PR DESCRIPTION
**Pull request by ncik-roberts**
The changes are slightly more than trivial (but only slightly). I've put them in a separate PR from #2820

### CODE REVIEW

File: ocaml/typing/printtyp.ml [hunk 1]
```diff
@@ -2428,37 +2290,15 @@ let class_type ppf cty =
   prepare_class_type [] cty;
   !Oprint.out_class_type ppf (tree_of_class_type Type [] cty)
 
-<<<<<<< HEAD
-let tree_of_class_param param var_inj =
```
Comment (goldfirere) [inline ocaml/typing/printtyp.ml:2432]: It looks like we're losing some logic here, but I did not look all that closely.

File: ocaml/typing/printtyp.ml [hunk 2]
```diff
@@ -3274,36 +3108,9 @@ let explain mis ppf =
...
-      try
-        ignore(Env.find_type p env : Types.type_declaration)
-      with Not_found ->
-=======
-    begin match Env.find_type p env with
-    | exception Not_found ->
->>>>>>> 5.2.0
-        fprintf ppf
-<<<<<<< HEAD
-          "@,@[<hov>Type %a was considered abstract@ when checking\
```
Comment (goldfirere) [inline ocaml/typing/printtyp.ml:3291]: It looks like we're losing this, but I don't think we should.

File: ocaml/typing/printtyp.ml (hunk 1 above)
Comment (ncik-roberts) [reply ocaml/typing/printtyp.ml:2432]: I think we're not losing any logic here.

File: ocaml/typing/printtyp.ml (hunk 2 above)
Comment (ncik-roberts) [reply ocaml/typing/printtyp.ml:3291]: I think we retain it below: https://github.com/ocaml-flambda/flambda-backend/pull/2821/files#diff-384fa66d4466a5b8c1e3dd8883e02c0c905aa4377e84be7c5afad25d0aa93cd5L3310-L3317

**Review by goldfirere — APPROVED** [review]

**Pull request merged**